tests/unit/test_memory_management.py

```
@@ -0,0 +1,341 @@
+        )
+        module = IRModule(name="test", functions=[func], type_definitions=[])
+
+        wrapper = gen.generate(module, "test")
+
+        # Should use 'is None' check, not falsy check
+        # This ensures Some(False) is not treated as None
+        lines = wrapper.split("\n")
+        found_is_none = False
+        for line in lines:
+            if "bool" in line and "if result is None:" in line:
+                found_is_none = True
```

Ruff (0.14.2)

Local variable `found_is_none` is assigned to but never used

Remove assignment to unused variable `found_is_none`

(F841)